CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -887,7 +887,7 @@ public static function processFormContribution(
 
     $isEmailReceipt = !empty($form->_values['is_email_receipt']);
     $isSeparateMembershipPayment = !empty($params['separate_membership_payment']);
-    $pledgeID = !empty($params['pledge_id']) ? $params['pledge_id'] : CRM_Utils_Array::value('pledge_id', $form->_values);
+    $pledgeID = !empty($params['pledge_id']) ? $params['pledge_id'] : $form->_values['pledge_id'] ?? NULL;
```

Comment:
This turns out to be ok no matter which has precedence, just might be slightly clearer with brackets around the last clause, i.e. `($form->_values['pledge_id'] ?? NULL)`